cli/internal/cloudcmd/create.go

```diff
@@ -215,14 +220,84 @@ func (c *Creator) createAzure(ctx context.Context, cl terraformClient, config *c
+	}
+
 	return clusterid.File{
-		CloudProvider: cloudprovider.Azure,
-		IP:            tfOutput.IP,
-		InitSecret:    []byte(tfOutput.Secret),
-		UID:           tfOutput.UID,
+		CloudProvider:  cloudprovider.Azure,
+		IP:             tfOutput.IP,
+		InitSecret:     []byte(tfOutput.Secret),
+		UID:            tfOutput.UID,
+		AttestationURL: tfOutput.AttestationURL,
 	}, nil
 }
 
+// PolicyPatcher interacts with Azure to update the attestation policy.
+type PolicyPatcher interface {
+	Patch(ctx context.Context, attestationURL, policy string) error
+}
+
+// NewPolicyPatcher returns a new PolicyPatcher.
+func NewPolicyPatcher() PolicyPatcher {
+	return policyPatcher{}
+}
+
+type policyPatcher struct{}
+
+// Patch updates the attestation policy to the base64-encoded attestation policy JWT for the given attestation URL.
+// https://learn.microsoft.com/en-us/azure/attestation/author-sign-policy#next-steps
+func (p policyPatcher) Patch(ctx context.Context, attestationURL, policy string) error {
+	// hacky way to update the MAA attestation policy. This should be changed as soon as either the Terraform provider supports it
+	// or the Go SDK gets updated to a recent API version.
+	// https://github.com/hashicorp/terraform-provider-azurerm/issues/20804
+	cred, err := azidentity.NewDefaultAzureCredential(nil)
+	if err != nil {
+		return fmt.Errorf("failed to create default Azure credentials: %w", err)
+	}
+	token, err := cred.GetToken(context.Background(), azpolicy.TokenRequestOptions{
+		Scopes: []string{"https://attest.azure.net/.default"},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create default Azure credentials: %w", err)
+	}
+
+	client := attestation.NewPolicyClient()
+
+	// azureGuest is the id for the "Azure VM" attestation type. Other types are documented here:
+	// https://learn.microsoft.com/en-us/rest/api/attestation/policy/set
+	req, err := client.SetPreparer(context.Background(), attestationURL, "azureGuest", p.encodeAttestationPolicy(constants.EncodedAzureMAAPolicy))
+	req.Header.Set("Authorization", "Bearer "+token.Token)
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	res, err := client.Send(req)
+	if err != nil {
+		return fmt.Errorf("failed to send request: %w", err)
+	}
+	defer res.Body.Close()
+
+	if res.StatusCode != http.StatusOK {
+		return fmt.Errorf("failed to update attestation policy: %s", res.Status)
+	}
+
+	return nil
+}
+
+// encodeAttestationPolicy encodes the base64-encoded attestation policy in the JWS format specified here:
+// https://learn.microsoft.com/en-us/azure/attestation/author-sign-policy#creating-the-policy-file-in-json-web-signature-format
+func (p policyPatcher) encodeAttestationPolicy(encodedPolicy string) string {
+	header := `{"alg":"none"}`
+	payload := fmt.Sprintf(`{"AttestationPolicy": "%s"}`, encodedPolicy)
+
+	encodedHeader := base64.RawStdEncoding.EncodeToString([]byte(header))
+	encodedPayload := base64.RawStdEncoding.EncodeToString([]byte(payload))
```

Comment:
```suggestion
	encodedHeader := base64.RawURLEncoding.EncodeToString([]byte(header))
	encodedPayload := base64.RawURLEncoding.EncodeToString([]byte(payload))
```
This must be url encoding. This may also be the problem you faced with the policy encoding.

Reply:
Agree! The issue with the policy encoding was a whitespace issue. I wasn't able to format the policy string so that the whitespace matched up with the one needed(?) by Azure? I tested some ways using backticks but didn't get it to work that way. (500 status code)